internal/controller/modelregistry_controller.go

```diff
@@ -1019,18 +1032,51 @@ func (r *ModelRegistryReconciler) setClusterDomain(ctx context.Context, registry
 	return nil
 }
 
-func (r *ModelRegistryReconciler) setDefaultCert(registry *modelregistryv1alpha1.ModelRegistry) bool {
+func (r *ModelRegistryReconciler) setDefaultCert(registry *modelregistryv1alpha1.ModelRegistry) (bool, error) {
 	updated := false
 	gateway := registry.Spec.Istio.Gateway
```

Comment:
We should be careful here because `registry.Spec.Istio` can be nil, so it would be safer to do a nil check to prevent a panic

Reply:
There is a guard earlier in [setRegistryDefaults()](https://github.com/opendatahub-io/model-registry-operator/blob/9d8f925deac08aa06f5648568dfbf082448b81ec/internal/controller/modelregistry_controller.go#L631) from where this is called.